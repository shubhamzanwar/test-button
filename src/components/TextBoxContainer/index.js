import React from 'react';
import TextBox from '../TextBox';
import { Button } from '../Button';

class TextBoxContainer extends React.Component{
    constructor(props){
        super(props);
        this.state = {
            text: '',
            counter: 0
        };
    }
    updateMessage = (message) => {
        this.setState({
            text: message
        });
    }
    countClick = () => {
        const {counter} = this.state;
        this.setState({
            counter: counter + 1
        });
    }
    render(){
        const {text, counter} = this.state;
        return(
            <div>
                <TextBox testID="test-input" type="text" onChange={(event) => {this.updateMessage(event)} }/>
                 <Button testID="test-btn" type="round" message={text} count={counter} onClick={this.countClick}/>
            </div>
        );
    }
}

export default TextBoxContainer;