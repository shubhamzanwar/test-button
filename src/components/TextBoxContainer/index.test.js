import React from 'react';
import TextBoxContainer from './index';
import { render, fireEvent } from '@testing-library/react';

describe('The TextBoxContainer Component', () => {
    it ('should render correctly', () => {
        const { asFragment } = render(<TextBoxContainer />);

        expect(asFragment()).toMatchSnapshot();
    });
    it ('should display the content in the texbox on the button', () => {
        const {getByTestId} = render(<TextBoxContainer/>);
        fireEvent.change(getByTestId("test-input"),{ target: { value:"sarvani"}});
        expect(getByTestId('test-btn')).toHaveTextContent("sarvani clicked 0 times")
    });
});