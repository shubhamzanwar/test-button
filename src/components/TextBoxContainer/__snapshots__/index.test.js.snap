// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The TextBoxContainer Component should render correctly 1`] = `
<DocumentFragment>
  <div>
    <div>
      <input
        data-testid="test-input"
        type="text"
      />
    </div>
    <button
      data-testid="test-btn"
    >
       clicked 0 times
    </button>
  </div>
</DocumentFragment>
`;
